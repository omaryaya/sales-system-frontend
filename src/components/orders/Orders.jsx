import React, { useEffect, useState } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { getOrders, deleteOrder, getOrderItems } from '../../actions/orders';
import OrdersTable from './OrdersTable';
import { Grid, Button } from '@material-ui/core';
import SimpleDialog from '../common/SimpleDialog';
import AddOrder from './AddOrder';
import OrderDetails from './OrderDetails';


const Orders = props => {

    const [isCreateOrderDialogOpen, setIsCreateOrderDialogOpen] = useState(false);

    useEffect(() => {
        props.getOrders();
    }, []);

    return (
        <div>
            
            <SimpleDialog component={AddOrder} open={isCreateOrderDialogOpen} title="Create Order" onClose={() => setIsCreateOrderDialogOpen(false)} />
            <Grid container spacing={2}>
                <Grid item xs={12}>
                    <h1>Orders</h1>
                    <Button variant="outlined" onClick={() => setIsCreateOrderDialogOpen(!isCreateOrderDialogOpen)}>
                        Create New Order
                    </Button>
                    <OrdersTable {...props}/>
                </Grid>
            </Grid>

        </div>
    );

}

Orders.propTypes = {
    orders: PropTypes.array.isRequired
}

const mapStateToProps = state => ({
    orders: state.orders.orders,
    /* content: state.orders.content,
    last: state.orders.last,
    page: state.orders.page,
    size: state.orders.size,
    totalElements: state.orders.totalElements,
    totalPages: state.orders.totalPages, */
})

export default connect(mapStateToProps, { getOrders, deleteOrder, getOrderItems })(Orders);






/*static propTypes = {
    orders: PropTypes.array.isRequired,
    content: PropTypes.array.isRequired,
    last: PropTypes.bool,
    page: PropTypes.number.isRequired,
    size: PropTypes.number,
    totalElements: PropTypes.number.isRequired,
    totalPages: PropTypes.number.isRequired,
}

const mapStateToProps = state => ({
    orders: state.orders.orders,
    content: state.orders.content,
    last: state.orders.last,
    page: state.orders.page,
    size: state.orders.size,
    totalElements: state.orders.totalElements,
    totalPages: state.orders.totalPages,
}) */