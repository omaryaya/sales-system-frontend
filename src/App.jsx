import React, { Component }/* , { useState, useEffect } */ from 'react';
import {
  BrowserRouter as Router,
  Switch, Route, Redirect
} from 'react-router-dom';
import './App.css';
import CssBaseline from '@material-ui/core/CssBaseline';
import Navbar from './components/Navbar';
import Products from './components/products/Products';
import Orders from './components/orders/Orders';

import { Provider } from 'react-redux';
import store from './store';
import Register from './components/accounts/Register';
import Login from './components/accounts/Login';
import PrivateRoute from './components/common/PrivateRoute';
import { PersistGate } from 'redux-persist/integration/react';
import { persistStore} from 'redux-persist';
// import Home from './components/Home';
// import { loadUser } from './actions/auth'
// import { render } from '@testing-library/react';

const persistor = persistStore(store);

class App extends Component {

  componentDidMount() {
    // store.dispatch(loadUser());
  }

  render() {
    return (

      <Provider store={store}>
        <PersistGate loading={null} persistor={persistor}>
        <Router>
          <CssBaseline />
          <Navbar />
          <div className="App">
            <Switch>
              <Route exact path="/login" component={Login} />
              <Route exact path="/register" component={Register} />
              <PrivateRoute exact path="/products" component={Products} />
              <PrivateRoute exact path="/orders" component={Orders} />
              
              <Redirect to="/login" />
            </Switch>
          </div>
        </Router>
        </PersistGate>
      </Provider>

    );

  }

}

export default App;
