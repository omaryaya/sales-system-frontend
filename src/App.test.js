import React from 'react';
import { render } from '@testing-library/react';
import App from './App';
import Orders from './components/orders/Orders';


import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import { Provider } from 'react-redux'
import { createStore } from 'redux'
import rootReducer from './reducers/index'

let wrap;
let store;

beforeEach(() => {
  store = createStore(rootReducer, {})
  wrap = shallow(<Provider store={store}><App/></Provider>)
})

it('renders without crashing', () => {
  
  console.debug(toJson(wrap));
  
})



test('empty test', () => {
  /* const { getByText } = render(<Orders />);
  const order = getByText(/orders/i);
  expect(order).toBeInTheDocument(); */
});
