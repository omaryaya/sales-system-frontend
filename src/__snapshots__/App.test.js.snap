// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders without crashing 1`] = `undefined`;

exports[`renders without crashing 1`] = `undefined`;
